consensus/blocks.go

```diff
@@ -107,13 +100,16 @@ func (s *State) validHeader(b Block) (err error) {
 		return EarlyTimestampErr
 	}
 
-	// Check that the block is not too far in the future.
-	skew := b.Timestamp - Timestamp(time.Now().Unix())
+	// Check that the block is not too far in the future. An external process
+	// will need to be responsible for resubmitting the block once it is no
+	// longer in the future.
+	skew := int(b.Timestamp) - int(Timestamp(time.Now().Unix()))
```

Comment:
honestly I would just keep them as `Timestamp`s and add a check for `b.Timestamp > Now`